libp2p/utils/semaphore.nim

```diff
@@ -77,13 +78,14 @@ proc release*(s: AsyncSemaphore) =
     trace "Releasing slot", available = s.count,
                             queue = s.queue.len
 
+    s.count.inc
```

Comment:
this doesn't guarantee that count goes above 0 - should it really complete queued futures then?

Reply:
That's how the incoming doesn't get starved: the slot "created" by a `forceAcquire` can be reused for incoming connections.
Of course, after a while this slot will be deleted (because more stuff will be released than `acquire`d)

Reply:
ok - hm, that is a little bit more permissive than I imagined it - it's a good point about starvation though, it does give more balance in return for longer periods of "non-compliance" - let's document  this in code in case it causes trouble we didn't anticipate